backend/services/feedback_service.py

```
@@ -108,4 +108,93 @@ def submit_new_grade_system(user_id: str, payload: dict):
+
+        if not isinstance(item_id, int) or isinstance(item_id, bool):
+            results.append({"itemType": item_type, "itemId": item_id, "ok": False, "error": "Invalid item id"})
+            continue
+
+        is_approved = action == 'approve'
+        proc = _APPROVAL_PROCS[item_type]
+        try:
+            with pool.connection() as conn, conn.transaction(), conn.cursor(row_factory=dict_row) as cur:
+                cur.execute(f"CALL {proc}(%s, %s, %s, %s)", (item_id, user_id, is_approved, None))
+            results.append({"itemType": item_type, "itemId": item_id, "ok": True, "action": action})
+        except Exception:
```

Ruff (0.15.15)

[warning] Do not catch blind exception: `Exception`

(BLE001)